docs/src/main/sphinx/develop/tests.md

```diff
@@ -0,0 +1,424 @@
+# Testing guidelines
+
+The Trino project aims to provide high quality artifacts and features to all
+users. A suite of tests used for development, continuous integration builds,
+and releases is critical for achieving this goal.
+
+The following requirements and guidelines help contributors to create tests with
+the following goals:
+
+* Fast execution.
+* Low requirements in terms of hardware, software, and compute power, and
+  therefore low cost.
+* Reliable execution and test results across multiple invocations.
+* Complex enough to test necessary aspects, while avoiding unnecessary
```

Comment:
I'd say tests should have enough coverage of execution paths without unnecessary complexity